src/include/sof/lib/ams.h

```diff
@@ -0,0 +1,242 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * Author: Krzysztof Frydryk <krzysztofx.frydryk@intel.com>
+ *
+ */
+
+#ifndef AM_SERVICE_H
+#define AM_SERVICE_H
+
+#include <errno.h>
+#include <sof/schedule/task.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/coherent.h>
+#include <sof/lib/uuid.h>
+
+#define AMS_INVALID_MSG_TYPE 0
+#define AMS_SERVICE_UUID_TABLE_SIZE 49
+#define AMS_INVALID_SLOT 0xFF
+#define AMS_ANY 0xFFFF
+
+#define AMS_ROUTING_TABLE_SIZE 41
```

Comment:
my gut feeling tells me, that 16 should be plenty and 4 will not be exceeded in the foreseeable future ;-)

Reply:
Fine, the values are totally artificial right now. I moved them to platform specific headers. I'm not really sure how to handle them, since they don't really depend on platform but rather on usecase (ie. how many modules do you want to communicate between). It's still probably better than hardcoding them there.